src/llmq/context.cpp

```diff
@@ -19,45 +19,53 @@
+llmq::CQuorumBlockProcessor* InitQuorumBlockProcessor(CEvoDB& evo_db, CConnman& connman, const std::unique_ptr<PeerManager>& peerman)
+{
+    llmq::quorumBlockProcessor = std::make_unique<llmq::CQuorumBlockProcessor>(evo_db, connman, peerman);
+    return llmq::quorumBlockProcessor.get();
 }
 
-LLMQContext::~LLMQContext() {
-    isman = nullptr;
-    clhandler = nullptr;
-    qman = nullptr;
-    quorum_block_processor = nullptr;
-
-    Destroy();
+llmq::CQuorumManager* InitQuorumManager(CEvoDB& evo_db, CConnman& connman,
+        CBLSWorker& bls_worker,
+        llmq::CDKGSessionManager& qdkgsman,
+        llmq::CQuorumBlockProcessor& quorum_block_processor,
+        const std::unique_ptr<PeerManager>& peerman)
+{
+    llmq::quorumManager = std::make_unique<llmq::CQuorumManager>(evo_db, connman, bls_worker, quorum_block_processor, qdkgsman, ::masternodeSync, peerman);
+    return llmq::quorumManager.get();
 }
 
-void LLMQContext::Create(CEvoDB& evo_db, CTxMemPool& mempool, CConnman& connman, CSporkManager& sporkman,
-                         const std::unique_ptr<PeerManager>& peerman, bool unit_tests, bool wipe) {
-    bls_worker = std::make_shared<CBLSWorker>();
+} // anonymous namespace
 
-    dkg_debugman = std::make_unique<llmq::CDKGDebugManager>();
-    llmq::quorumBlockProcessor = std::make_unique<llmq::CQuorumBlockProcessor>(evo_db, connman, peerman);
-    qdkgsman = std::make_unique<llmq::CDKGSessionManager>(connman, *bls_worker, *dkg_debugman, *llmq::quorumBlockProcessor, sporkman, peerman, unit_tests, wipe);
-    llmq::quorumManager = std::make_unique<llmq::CQuorumManager>(evo_db, connman, *bls_worker, *llmq::quorumBlockProcessor, *qdkgsman, ::masternodeSync, peerman);
-    sigman = std::make_unique<llmq::CSigningManager>(connman, *llmq::quorumManager, peerman, unit_tests, wipe);
-    shareman = std::make_unique<llmq::CSigSharesManager>(connman, *llmq::quorumManager, *sigman, peerman);
+LLMQContext::LLMQContext(CEvoDB& evo_db, CTxMemPool& mempool, CConnman& connman, CSporkManager& sporkman,
+                         const std::unique_ptr<PeerManager>& peerman, bool unit_tests, bool wipe) :
+    bls_worker{std::make_shared<CBLSWorker>()},
+    dkg_debugman{std::make_unique<llmq::CDKGDebugManager>()},
+    quorum_block_processor{InitQuorumBlockProcessor(evo_db, connman, peerman)},
+    qdkgsman{std::make_unique<llmq::CDKGSessionManager>(connman, *bls_worker, *dkg_debugman, *quorum_block_processor, sporkman, peerman, unit_tests, wipe)},
+    qman{InitQuorumManager(evo_db, connman, *bls_worker, *qdkgsman, *quorum_block_processor, peerman)},
+    sigman{std::make_unique<llmq::CSigningManager>(connman, *llmq::quorumManager, peerman, unit_tests, wipe)},
+    shareman{std::make_unique<llmq::CSigSharesManager>(connman, *llmq::quorumManager, *sigman, peerman)}
+{
     llmq::chainLocksHandler = std::make_unique<llmq::CChainLocksHandler>(mempool, connman, sporkman, *sigman, *shareman, *llmq::quorumManager, *::masternodeSync, peerman);
```

Comment:
What are the circumstances behind `llmq::quorumBlockProcessor` being initialized through a function while `llmq::chainLocksHandler` is initialized through the initializer body?

Reply:
`quorumBlockProcessor` are used in other members initialization, for example `qman` and `shareman` depends on then.

But `chainLocksHandler` are not used anyhow to initialize other members and it is a global variable.
So, it doesn't actually important how exactly to initialized.

Btw, thanks for comment. I unified implementation due to 2 reasons:
 - to add `const` for all members to be sure that no one will externally change these pointers in LLMQContext
 - for simplifying code - you don't need to carefully check all dependencies inside LLMQContext to understand why there's different way to do same things.